base/gmp.jl

```diff
@@ -416,6 +416,10 @@ function ^(x::BigInt, y::Culong)
     ccall((:__gmpz_pow_ui, :libgmp), Void, (Ptr{BigInt}, Ptr{BigInt}, Culong), &z, &x, y)
     return z
 end
+@generated function ^{p}(x::BigInt, ::Type{Val{p}})
+    p < 0 && return :(inv(x)^p)
```

Comment:
This really need not be a generated function. Constant folding should be able to take care of this just fine.